src/transformers/pipelines/__init__.py

```diff
@@ -785,6 +789,10 @@ def pipeline(
                 **hub_kwargs,
```

Comment:
Why not just:

```suggestion
                cache_dir=model_kwargs.get("cache_dir"),
                **hub_kwargs,
```

Reply:
Would that work properly if there is no cache_dir in model_kwargs? Keep in mind that I've learned and forgotten Python twice, and I guess I'm on round 3.

Reply:
Yep! `get` is the safe way to access a value from a dictionary e.g.:

```
d = {'a': 1, 'b': 2, 'c': 3}

# This fails
d['d']

# This returns None
d.get('d')
```